src/components/RequestList/RequestItem/index.tsx

```diff
@@ -68,7 +77,30 @@ const RequestItem: React.FC<RequestItemProps> = ({ request, onDelete }) => {
   const deleteRequest = async () => {
     await axios.delete(`/api/v1/request/${request.id}`);
 
-    onDelete();
+    revalidateList();
+  };
+
+  const retryRequest = async () => {
+    setRetrying(true);
+
+    const requestPayload =
+      request.type === MediaType.MOVIE
+        ? { mediaId: request.media.tmdbId, mediaType: MediaType.MOVIE }
+        : {
+            mediaId: request.media.tmdbId,
+            tvdbId: request.media.tvdbId,
+            mediaType: MediaType.TV,
+            seasons: request.media.seasons,
+          };
+    try {
+      await axios.delete(`/api/v1/request/${request.id}`);
+      await axios.post<MediaRequest>(`/api/v1/request`, requestPayload);
```

Comment:
If someone else requests this, and it fails. Won't this then change the creator of the request to the admin who retries?

There is an endpoint to actually update the status of a request back to approved. That _should_ actually try to send the request again. It will also prevent the request from jumping in the list from whatever page they are on back to the top.

Reply:
I ended up doing two requests, one to set it to pending and another to set it to approved, so it actually triggers the AfterUpdate hook. This should probably be a dedicated endpoint in the next iteration on this I think.